github2gitea-mirror

```diff
@@ -163,8 +194,27 @@ create_migration_repo() {
     log "Create migration repo."
     if ! $CURL -w  "%{http_code}\n"  "${header_options[@]}" -d @- -X POST $GITEA_URL/api/v1/repos/migrate > ${jsonoutput}/result.txt 2>${jsonoutput}/stderr.txt; then
         local code=$(<${jsonoutput}/result.txt)
-        if (( code != 409 ));then # 409 == repo already exits
+        if (( code == 409 ));then # 409 == repo already exits
+            log "409: Gitea repo already exists. Skipping migration."
+        else
+            cat ${jsonoutput}/stderr.txt >&2
+            exit 1
+        fi
+    fi
+}
+
+# Creates a specific public/private user on Gitea
+create_migration_user() {
+    visibility="${1:-}"
+    log "Create migration user with name: ${github_user}"
+    local password=$(tr -dc 'A-Za-z0-9!@#$%^&*()_+\-=[]{}\|;:,.<>/`~' < /dev/urandom | head -c 64)
+    if ! $CURL -X POST $GITEA_URL/api/v1/admin/users "${header_options[@]}" --data '{"username": "'"${github_user}"'", "visibility": "'"${visibility}"'", "email": "'"${github_user}@example.com"'", "password": "'"${password}"'", "must_change_password": true}' > ${jsonoutput}/result.txt 2>${jsonoutput}/stderr.txt; then
```

Comment:
Despite the Gitea API documentation saying the `password` field is not required, it _is_ required, and Gitea will return a 400 Bad Request error without it.